src/main/java/org/apache/commons/compress/archivers/zip/ZipFile.java

```diff
@@ -1111,14 +1122,11 @@ public int read() throws IOException {
                 }
                 return -1;
             }
-            synchronized (archive) {
-                archive.position(loc++);
-                int read = read(1);
-                if (read < 0) {
-                    return read;
-                }
-                return buffer.get() & 0xff;
+            int read = read(loc++, 1);
+            if (read < 0) {
```

Comment:
I think it depends on what we think `synchronized` is supposed to protect against.

As it stands the `synchronized` block also protected concurrent reads from the same `BoundedInputStream` from overwriting their results (protecting `loc` and `buffer` not just the file position).

If we want to keep that level of protection the changes are not going to work. If we say each input stream returned by `getInputStream` can only be used by a single thread, then I think it is fine to keep the increment outside of the block and only synchronise the code that protects the interaction between different streams.